features/frameworks/greentea-client/source/test_env.cpp

```diff
@@ -61,8 +61,10 @@ static void greentea_notify_version();
  * Rawserial object used to provide direct, raw serial communications
  * between the target and the host.
  */
-RawSerial greentea_serial(USBTX, USBRX); 
-
+RawSerial& greentea_serial() { 
+    static RawSerial serial(USBTX, USBRX);
```

Comment:
>Since these both solve the same problem of global-initialization, shouldn't we chose one solution and consolidate?

👍 

>I would be all for getting rid of the SingletonPtr class. It is not very intuitive to declare.

Though this was to be renamed to LazyInitialization

> Only downside is it won't have the semantics of pointer access to a global class.

The semantics of returning a reference and not being able to overloading the dot operator is very unintuitive. Pointer access to a global class is a HUGE 👍  for readability and clarity IMO.